src/frontends/onnx/frontend/src/core/tensor.hpp

```diff
@@ -457,9 +476,47 @@ class Tensor {
     }
 
 private:
-    template <typename T>
+    template <typename T,
+              typename std::enable_if<std::is_same<T, float>::value || std::is_same<T, double>::value ||
+                                          std::is_same<T, int32_t>::value || std::is_same<T, int64_t>::value ||
+                                          std::is_same<T, uint64_t>::value,
+                                      bool>::type = true>
+    std::shared_ptr<ngraph::op::Constant> make_ng_constant(const element::Type& type) const {
+        std::shared_ptr<default_opset::Constant> constant{nullptr};
+        int data_size = detail::get_data_size(*m_tensor_proto);
+        if (detail::has_tensor_external_data(*m_tensor_proto)) {
+            auto external_data = detail::load_external_data(*m_tensor_proto);
+            constant = std::make_shared<ngraph::op::Constant>(type, m_shape, external_data.data());
+        } else if (data_size == shape_size(m_shape)) {
+            constant = std::make_shared<ngraph::op::Constant>(type, m_shape, detail::get_data_ptr(*m_tensor_proto));
+        } else if (data_size == 0 && m_shape.size() == 0) {
+            constant = std::make_shared<default_opset::Constant>(type, Shape{}, 0);
+        } else {
+            throw error::tensor::shape_doesnt_match_data_size{};
+        }
+
+        if (m_tensor_proto->has_name()) {
+            constant->set_friendly_name(get_name());
+        }
+        return constant;
+    }
+
+    template <typename T,
+              typename std::enable_if<!std::is_same<T, float>::value && !std::is_same<T, double>::value &&
+                                          !std::is_same<T, int32_t>::value && !std::is_same<T, int64_t>::value &&
+                                          !std::is_same<T, uint64_t>::value,
+                                      bool>::type = true>
     std::shared_ptr<ngraph::op::Constant> make_ng_constant(const element::Type& type) const {
-        auto constant = std::make_shared<ngraph::op::Constant>(type, m_shape, get_data<T>());
+        std::shared_ptr<default_opset::Constant> constant{nullptr};
+        auto data = get_data<T>();
+        auto data_size = data.size();
+        if (data_size == shape_size(m_shape)) {
+            constant = std::make_shared<ngraph::op::Constant>(type, m_shape, get_data<T>());
```

Comment:
Shouldn't you pass the `data` variable here instead of calling `get_data<T>` again?
```suggestion
            constant = std::make_shared<ngraph::op::Constant>(type, m_shape, data);
```

Reply:
done